test/jdk/sun/security/validator/CertReplace.java

```diff
@@ -26,22 +26,190 @@
+                              "-genkeypair -alias " + caAlias + " -dname CN=CA -keyalg rsa -sigalg md2withrsa -ext bc");
+        SecurityTools.keytool(ktBaseParameters +
+                              "-genkeypair -alias " + intAliase + " -dname CN=Int -keyalg rsa");
+        SecurityTools.keytool(ktBaseParameters +
+                              "-genkeypair -alias " + userAlias + " -dname CN=User -keyalg rsa");
+
+        final KeyStore keyStore = KeyStoreUtils.loadKeyStore(CERTREPLACE_JKS, PASSWORD);
+
+        // 2. Signing: ca -> int -> user
+        final String intReqFile = intAliase + ".req";
+        final String intCertFileName = intAliase + ".cert";
+
+        SecurityTools.keytool(ktBaseParameters +
+                              "-certreq -alias " + intAliase + " -file " + intReqFile);
+        SecurityTools.keytool(ktBaseParameters +
+                              "-gencert -rfc -alias " + caAlias + " -ext bc -infile " + intReqFile + " " +
+                              "-outfile " + intCertFileName);
+
+        //putting the certificate in the keystore
+        final CertificateFactory certificateFactory = CertificateFactory.getInstance("X509");
+        final Certificate[] certs = new Certificate[]{
+                certificateFactory.generateCertificate(
+                        new FileInputStream(intCertFileName)
+                )
+        };
+        final PrivateKey privateKey = (PrivateKey) keyStore.getKey(intAliase, PASSWORD.toCharArray());
+        keyStore.setKeyEntry(intAliase, privateKey, PASSWORD.toCharArray(), certs);
+        keyStore.store(new FileOutputStream(CERTREPLACE_JKS), PASSWORD.toCharArray());
+
+
+        final String userReqFile = userAlias + ".req";
+        SecurityTools.keytool(ktBaseParameters +
+                              "-certreq -alias " + userAlias + " -file " + userReqFile);
+        SecurityTools.keytool(ktBaseParameters +
+                              "-gencert -rfc -alias " + intAliase + " " +
+                              "-infile " + userReqFile + " " +
+                              "-outfile " + certplaceCerts); // this will create certreplace.certs which is later appended
+
+        // 3. Create the certchain file
+        final Path certPath = Paths.get(certplaceCerts);
+
+        Files.write(certPath, Files.readAllBytes(Path.of(intCertFileName)), StandardOpenOption.APPEND);
+
+        final String outputCa = SecurityTools.keytool(ktBaseParameters +
+                                                      "-export -rfc -alias " + caAlias).getOutput();
+        Files.write(certPath, outputCa.getBytes(), StandardOpenOption.APPEND);
+
+        // 4. Upgrade ca from MD2withRSA to SHA256withRSA, remove other aliases and make this keystore the cacerts file
+        SecurityTools.keytool(ktBaseParameters +
+                              "-selfcert -alias " + caAlias);
+        keyStore.deleteEntry(intAliase);
+        keyStore.deleteEntry(userAlias);
```

Comment:
Call `keyStore.store` to actually remove the 2 entries inside the keystore file. But this needs to be done before the `-selfcert` call, otherwise, the old cert would overwrite the new one.